src/js/background/background.js

```diff
@@ -274,6 +287,88 @@ async function refreshAccountPages() {
   });
 }
 
+// This function is defined as global in the ESLint config _because_ it is created here:
+// eslint-disable-next-line no-redeclare
+async function makeDomainAddress(address, block_list_emails, description = null) {
+  const apiToken = await browser.storage.local.get("apiToken");
+
+  if (!apiToken.apiToken) {
+    browser.tabs.create({
+      url: RELAY_SITE_ORIGIN,
+    });
+    return;
+  }
+
+  const { relayApiSource } = await browser.storage.local.get("relayApiSource");  
+  const serverStoragePermission = await getServerStoragePref();
+  const relayApiUrlRelayAddress = `${relayApiSource}/domainaddresses/`;
+
+  let apiBody = {
+    "enabled": true,
+    "description": "",
+    "block_list_emails": block_list_emails,
+    "used_on": "",
+    "address": address,
+  };
+
+  // Only send description/generated_for/used_on fields in the request if the user is opt'd into server storage
+  if (description && serverStoragePermission) {
+    apiBody.description = description;
+    apiBody.generated_for = description;
+    apiBody.used_on = description + ",";
+  }
+
+
+  const headers = await createNewHeadersObject({auth: true});
+
+  const newRelayAddressResponse = await fetch(relayApiUrlRelayAddress, {
+    mode: "same-origin",
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify(apiBody),
+  });
+
+  if (newRelayAddressResponse.status === 402) {
+    // FIXME: can this just return newRelayAddressResponse ?
+    return { status: 402 };
+  }
+
+  if (newRelayAddressResponse.status === 409) {
+    return { status: 409 };
+  }
+  
+  if (newRelayAddressResponse.status === 400) {
+    return { status: 400 };
+  }
+
+  let newRelayAddressJson = await newRelayAddressResponse.json();
+
+  if (description) {
+    // TODO: Update the domain attribute to be "label"
+    newRelayAddressJson.description = description;
+    // Store the domain in which the alias was generated, separate from the label
+    newRelayAddressJson.generated_for = description;
+  }
+
+  // TODO: put this into an updateLocalAddresses() function
```

Comment:
I moved this into a helper function. Revised in fe9a02f32e845b816af80dab964dab630b83a7d2